src/transformers/pipelines/base.py

```diff
@@ -1015,9 +1015,17 @@ def postprocess(self, model_outputs: ModelOutput, **postprocess_parameters: Dict
         raise NotImplementedError("postprocess not implemented")
 
     def get_inference_context(self):
+        torch_version = version.parse(torch.__version__).base_version
+        inference_mode_supported = version.parse(torch_version) >= version.parse("1.9.0")
+        try:
+            from torch._dynamo.eval_frame import OptimizedModule
```

Comment:
Here I'd prefer a version check (with 2.0) since I'm afraid the PyTorch team might move this class in another module in the future (current one being private and all)

Reply:
But version check is not enough, we need to check if the model was compiled or not

Reply:
Yes but version check is enough to import this instead of the try/except.

Reply:
Given those concerns, I'd vote to simply drop `inference_mode`. It does not even help that much.
wdyt ?

Reply:
I have no concerns dropping it completely. I wasn't the one who introduced it, so wasn't sure what it makes us gain.

Reply:
I'll remove it then

Reply:
At the time, it was the new thing from pytorch to make inference faster, some models had a ~5% improvement iirc.